src/extensions/step_handlers/script_handler/src/script_handler.cpp

```diff
@@ -544,22 +560,40 @@ static ADUC_Result ScriptHandler_PerformAction(const std::string& action, const
     // selected component, we will skip the 'apply' phase, and then skip the
     // remaining install-item(s).
     // Also, don't continue if WorkflowHandle is NULL in the ADUInterface_Connected->HandleStartupWorkflowData flow.
-    if (result.ResultCode == ADUC_Result_Install_Skipped_UpdateAlreadyInstalled
+    if (results.result.ResultCode == ADUC_Result_Install_Skipped_UpdateAlreadyInstalled
         || workflowData->WorkflowHandle == nullptr)
     {
         goto done;
     }
 
     aduShellArgs.emplace_back(adushconst::target_data_opt);
-    aduShellArgs.emplace_back(scriptFilePath);
+    aduShellArgs.emplace_back(results.scriptFilePath);
+    results.commandLineArgs.emplace_back(results.scriptFilePath);
+
+    // Prepare arguments based on specified api version.
+    if (apiVer == nullptr || strcmp(apiVer, "1.0") == 0)
```

Comment:
still not thrilled with using string comparison for versions.  can this not be an integer?